### ISSUE TITLE
Certificates crossing the CRITICAL threshold are not flagged as CRITICAL state

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-cert#188]
I believe this works as intended, but when reviewing this project as part of work on a new one this block stuck out:

https://github.com/atc0005/check-cert/blob/999506d4ce0ca3f33f017dd356b74f02fab23649/cmd/check_cert/main.go#L222-L266

At a glance, this looks like crossing the CRITICAL age threshold will only result in a WARNING state up until the certificate expires. I'm pretty sure this isn't the case though, but it's worth a more thorough review just to be sure.

**Comment by atc0005** [comment]
Also relevant:

https://github.com/atc0005/check-cert/blob/999506d4ce0ca3f33f017dd356b74f02fab23649/internal/certs/certs.go#L810-L820

Two things:

- Since we're setting `ServiceCheckStatus`, we should probably also add a `ServiceCheckCode` field
- The `ServiceCheckStatus` is also hard-coded to use `nagios.StateWARNINGLabel`
  - this should instead be based on whether the `CRITICAL` threshold has been crossed
    - if chain has expiring certs and has not crossed `CRITICAL` threshold, then the state is `WARNING`
    - if chain has expiring certs and *has* crossed `CRITICAL` threshold, then the state is `CRITICAL`

**Comment by atc0005** [comment]
Quick test results below. In short, we have a problem.

## Test `WARNING` state

The google.com cert has 53d, 15h remaining. The options specified here should trigger a `WARNING` state.

```shell
$ ./check_cert-v0.4.2-0-gd2cd8cc-linux-amd64 --server www.google.com --age-critical 50  --age-warning 60
{"level":"warn","version":"check-cert v0.4.2-0-gd2cd8cc (https://github.com/atc0005/check-cert)","logging_level":"info","app_type":"plugin","cert_check_timeout":"10s","age_warning":60,"age_critical":50,"expected_sans_entries":"","server":"www.google.com","port":443,"error":"1 certificates expired or expiring","expiring_certs":1,"time":"2021-07-28T08:09:45-05:00","caller":"github.com/atc0005/check-cert/cmd/check_cert/main.go:246","message":"expiring certs present in chain"}
WARNING: leaf cert "www.google.com" expires next with 53d 15h remaining (until 2021-09-20 04:12:57 +0000 UTC) [EXPIRED: 0, EXPIRING: 1, OK: 2]

**ERRORS**

* 1 certificates expired or expiring

**THRESHOLDS**

* CRITICAL: Expires before 2021-09-16 13:09:45 +0000 UTC (50 days)
* WARNING: Expires before 2021-09-26 13:09:45 +0000 UTC (60 days)

**DETAILED INFO**

Certificate 1 of 3 (leaf):
        Name: CN=www.google.com
        SANs entries: [www.google.com]
        KeyID: 69:C1:E8:D:70:84:80:3E:6F:E3:78:D7:A3:52:C5:81:A:51:CE:89
        Issuer: CN=GTS CA 1C3,O=Google Trust Services LLC,C=US
        IssuerKeyID: 8A:74:7F:AF:85:CD:EE:95:CD:3D:9C:D0:E2:46:14:F3:71:35:1D:27
        Serial: 12:D4:D6:BA:D3:7B:1D:D1:0A:00:00:00:00:EB:61:08
        Issued On: 2021-06-28 04:12:58 +0000 UTC
        Expiration: 2021-09-20 04:12:57 +0000 UTC
        Status: [WARNING] 53d 15h remaining

Certificate 2 of 3 (intermediate):
        Name: CN=GTS CA 1C3,O=Google Trust Services LLC,C=US
        SANs entries: []
        KeyID: 8A:74:7F:AF:85:CD:EE:95:CD:3D:9C:D0:E2:46:14:F3:71:35:1D:27
        Issuer: CN=GTS Root R1,O=Google Trust Services LLC,C=US
        IssuerKeyID: E4:AF:2B:26:71:1A:2B:48:27:85:2F:52:66:2C:EF:F0:89:13:71:3E
        Serial: 02:03:BC:53:59:6B:34:C7:18:F5:01:50:66
        Issued On: 2020-08-13 00:00:42 +0000 UTC
        Expiration: 2027-09-30 00:00:42 +0000 UTC
        Status: [OK] 2254d 10h remaining

Certificate 3 of 3 (intermediate):
        Name: CN=GTS Root R1,O=Google Trust Services LLC,C=US
        SANs entries: []
        KeyID: E4:AF:2B:26:71:1A:2B:48:27:85:2F:52:66:2C:EF:F0:89:13:71:3E
        Issuer: CN=GlobalSign Root CA,OU=Root CA,O=GlobalSign nv-sa,C=BE
        IssuerKeyID: 60:7B:66:1A:45:D:97:CA:89:50:2F:7D:4:CD:34:A8:FF:FC:FD:4B
        Serial: 77:BD:0D:6C:DB:36:F9:1A:EA:21:0F:C4:F0:58:D3:0D
        Issued On: 2020-06-19 00:00:42 +0000 UTC
        Expiration: 2028-01-28 00:00:42 +0000 UTC
        Status: [OK] 2374d 10h remaining
```

Summary:

- Status label for cert chain entries is set properly
- Exit code is `1`, which maps to `nagios.StateWARNINGExitCode`. This is good.
- State label is `WARNING`. This is good.

## Test `CRITICAL` state

The google.com cert has 53d, 15h remaining. The options specified here should trigger a `CRITICAL` state.

```shell
$ ./check_cert-v0.4.2-0-gd2cd8cc-linux-amd64 --server www.google.com --age-critical 60  --age-warning 90
{"level":"warn","version":"check-cert v0.4.2-0-gd2cd8cc (https://github.com/atc0005/check-cert)","logging_level":"info","app_type":"plugin","cert_check_timeout":"10s","age_warning":90,"age_critical":60,"expected_sans_entries":"","server":"www.google.com","port":443,"error":"1 certificates expired or expiring","expiring_certs":1,"time":"2021-07-28T08:11:10-05:00","caller":"github.com/atc0005/check-cert/cmd/check_cert/main.go:246","message":"expiring certs present in chain"}
WARNING: leaf cert "www.google.com" expires next with 53d 15h remaining (until 2021-09-20 04:12:57 +0000 UTC) [EXPIRED: 0, EXPIRING: 1, OK: 2]

**ERRORS**

* 1 certificates expired or expiring

**THRESHOLDS**

* CRITICAL: Expires before 2021-09-26 13:11:10 +0000 UTC (60 days)
* WARNING: Expires before 2021-10-26 13:11:10 +0000 UTC (90 days)

**DETAILED INFO**

Certificate 1 of 3 (leaf):
        Name: CN=www.google.com
        SANs entries: [www.google.com]
        KeyID: 69:C1:E8:D:70:84:80:3E:6F:E3:78:D7:A3:52:C5:81:A:51:CE:89
        Issuer: CN=GTS CA 1C3,O=Google Trust Services LLC,C=US
        IssuerKeyID: 8A:74:7F:AF:85:CD:EE:95:CD:3D:9C:D0:E2:46:14:F3:71:35:1D:27
        Serial: 12:D4:D6:BA:D3:7B:1D:D1:0A:00:00:00:00:EB:61:08
        Issued On: 2021-06-28 04:12:58 +0000 UTC
        Expiration: 2021-09-20 04:12:57 +0000 UTC
        Status: [CRITICAL] 53d 15h remaining

Certificate 2 of 3 (intermediate):
        Name: CN=GTS CA 1C3,O=Google Trust Services LLC,C=US
        SANs entries: []
        KeyID: 8A:74:7F:AF:85:CD:EE:95:CD:3D:9C:D0:E2:46:14:F3:71:35:1D:27
        Issuer: CN=GTS Root R1,O=Google Trust Services LLC,C=US
        IssuerKeyID: E4:AF:2B:26:71:1A:2B:48:27:85:2F:52:66:2C:EF:F0:89:13:71:3E
        Serial: 02:03:BC:53:59:6B:34:C7:18:F5:01:50:66
        Issued On: 2020-08-13 00:00:42 +0000 UTC
        Expiration: 2027-09-30 00:00:42 +0000 UTC
        Status: [OK] 2254d 10h remaining

Certificate 3 of 3 (intermediate):
        Name: CN=GTS Root R1,O=Google Trust Services LLC,C=US
        SANs entries: []
        KeyID: E4:AF:2B:26:71:1A:2B:48:27:85:2F:52:66:2C:EF:F0:89:13:71:3E
        Issuer: CN=GlobalSign Root CA,OU=Root CA,O=GlobalSign nv-sa,C=BE
        IssuerKeyID: 60:7B:66:1A:45:D:97:CA:89:50:2F:7D:4:CD:34:A8:FF:FC:FD:4B
        Serial: 77:BD:0D:6C:DB:36:F9:1A:EA:21:0F:C4:F0:58:D3:0D
        Issued On: 2020-06-19 00:00:42 +0000 UTC
        Expiration: 2028-01-28 00:00:42 +0000 UTC
        Status: [OK] 2374d 10h remaining
```

In short, problem confirmed.

Summary:

- Status label for cert chain entries is set properly
- Exit code is `1`, which maps to `nagios.StateWARNINGExitCode`. This is wrong, it should be `2`, or `nagios.StateCRITICALExitCode`
- State label is `WARNING`. This is wrong. It should be `CRITICAL`

**Comment by atc0005** [comment]
Haven't checked the scope of the problem. I suspect that at minimum `lscert` is affected too, but haven't looked yet.

**Comment by atc0005** [comment]
Invalid logic fixed in GH-189, but leaving this open until I can refresh the README examples to reflect the fixed logic.
